internal/backend/db/dbgorm/events.go

```diff
@@ -22,13 +22,19 @@ func (db *gormdb) SaveEvent(ctx context.Context, event sdktypes.Event) error {
 
 	cid := event.ConnectionID() // could be invalid/zero
 
+	data, cdata, err := db.compressJSON(event.Data())
+	if err != nil {
+		return err
+	}
+
 	e := scheme.Event{
-		EventID:      event.ID().UUIDValue(),
-		ConnectionID: scheme.UUIDOrNil(cid.UUIDValue()),
-		EventType:    event.Type(),
-		Data:         kittehs.Must1(json.Marshal(event.Data())),
-		Memo:         kittehs.Must1(json.Marshal(event.Memo())),
-		CreatedAt:    event.CreatedAt(),
+		EventID:        event.ID().UUIDValue(),
+		ConnectionID:   scheme.UUIDOrNil(cid.UUIDValue()),
+		EventType:      event.Type(),
+		Data:           data,
+		CompressedData: cdata,
```

Comment:
Use the same field for `Data` and have a `Compressed` flag.
This can also changed the signature of `compressJSON` to return data and a flag.